lms/services/canvas_api.py

```diff
@@ -27,15 +28,24 @@ def get_token(self, authorization_code):
         Get an access token for the current LTI user.
 
         Posts to the Canvas API to get the access token and returns it.
+
+        :arg authorization_code: The Canvas API OAuth 2.0 authorization code to
+            exchange for an access token
+        :type authorization_code: str
+
+        :raise lms.services.CanvasAPIServerError: if the Canvas API request
+            fails for any reason
         """
         access_token_request = self._helper.access_token_request(authorization_code)
-        access_token_response = requests.Session().send(access_token_request)
 
-        # TODO: Handle access token error responses.
-        # These aren't documented in the Canvas docs as far as I can see.
-        # They are documented in the OAuth 2 spec but I don't yet know if Canvas's
-        # error responses follow this:
-        # https://tools.ietf.org/html/rfc6749#section-5.2
+        try:
+            access_token_response = requests.Session().send(access_token_request)
+            access_token_response.raise_for_status()
+        except RequestException as err:
+            raise CanvasAPIServerError(
+                explanation="Authorizing with Canvas failed",
+                response=getattr(err, "response", None),
+            ) from err
```

Comment:
This catches all kinds of requests exceptions (network timeouts and things where we don't even get a response, as well as non-2xx responses, etc). It's the same technique we figured out for when the LMS app calls the h API